### PR TITLE
[#593] Fix URI handling bug

### PR DESCRIPTION
**Pull request by jfacorro**
### Description

Fix URI handling bug.

Fixes #593.

### CODE REVIEW

**Comment by robertoaloi** [comment]
Merging this since the current version breaks on Mac OS. @filmor could you check it works on Windows, too? We can refine it after-merge, if it does not.

**Comment by filmor** [comment]
It will not work on Windows. The slicing is required because the path component will look like `/c:/some_thing` with is not a valid path. I'll look into it again.

File: src/els_uri.erl [hunk 1]
```diff
@@ -52,7 +52,7 @@ uri(Path) ->
            true -> els_utils:to_binary(string:replace(Path, "\\", "/"));
            false -> Path
          end,
-  <<"file:///", Path1/binary>>.
+  <<"file://", Path1/binary>>.
```
Comment (filmor) [inline src/els_uri.erl]: This is definitely wrong if you don't guarantee an absolute path.

Comment (jfacorro) [reply src/els_uri.erl]: I may be wrong, but as far as I could see, the clients always send absolute paths to the server. Isn't this the case?

Comment (filmor) [reply src/els_uri.erl]: It is, but this is the other way round (from Erlang to client). I've found the "official" logic and will try to implement it here: https://github.com/microsoft/vscode-uri/blob/master/src/index.ts#L1005-L1026

**Pull request merged**